Blocks/URL/url.h

```diff
@@ -445,6 +447,64 @@ inline URLPathArgs::CountMask& operator<<=(URLPathArgs::CountMask& lhs, const si
   return lhs;
 }
 
+
+constexpr char URLPathSeparator = '/';
+
+inline char GetURLPathSeparator() {
+  return URLPathSeparator;
+}
+
+
+inline std::string JoinURLPath(const std::string& left, const std::string& right) {
+  std::string left_trimmed = left;
+  std::string right_trimmed = right;
+
+  // NOTE(sompylasar): The only case we need to re-add the slash at the left is the root URL case.
+  const bool is_root = (left.length() == 1 && left.front() == URLPathSeparator);
+
+  // "/" -> ""; "////" -> ""; "/xxx/" -> "/xxx"; "xxx/" -> "xxx";
+  left_trimmed.erase(left_trimmed.find_last_not_of(URLPathSeparator) + 1);  // trim right
+
+  // "xxx" -> "xxx"; "/xxx" -> "xxx"; "////xxx" -> "xxx";
+  right_trimmed.erase(0, right_trimmed.find_first_not_of(URLPathSeparator));  // trim left
+
+  std::string joined;
+
+  if (is_root) {
+    joined += URLPathSeparator;
+  }
+
+  joined += left_trimmed;
+
+  if (!left_trimmed.empty() && !right_trimmed.empty()) {
+    joined += URLPathSeparator;
+  }
+
+  joined += right_trimmed;
+
+  return joined;
+}
+
+
+struct ConvertFileSystemPathToURLPathDirectoryMismatchException : current::Exception {
```

Comment:
Why do you even need this? Recursive scan dir tells you which directory did it enter. Join then by the forward slash to get relative URLs.

You could add a `vector<string>` or `set<string>` to recursive dir scanner (by const reference, of course), and `current::strings::Join` can accept a container.

Manipulating paths, especially in non-OS-agnostic manner, is sure not the backend's job.